### PR TITLE
split routes into multiple files + make custom auth middleware

### PR DESCRIPTION
**Pull request by dsifford**
This PR does 2 things:

1. Splits the routes into multiple files to make it easier to maintain.
2. Introduces a custom auth middleware that checks JWT and requires `private_key` header (which subsequently adds the value of that to `req.privateKey` for easy consumption.).

Let me know if you have any questions.

### CODE REVIEW

File: middleware/lib/middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Auth0 JWT authentication middleware.
+ */
+const checkJwt = jwt({
+  secret: jwksRsa.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksUri: `${AUTH0_BASE}/.well-known/jwks.json`,
+  }),
+  audience: `${AUTH0_BASE}/api/v2/`,
+  issuer: AUTH0_BASE,
```
Comment (dsifford) [inline middleware/lib/middleware/auth.js:19]: @dbartalino Was the trailing slash important here? If so, we'll need to add that back in.

File: middleware/package.json [hunk 2]
```diff
@@ -15,7 +15,6 @@
     "dotenv": "^6.2.0",
     "express": "^4.16.4",
     "express-jwt": "^5.3.1",
-    "express-jwt-authz": "^1.0.0",
```
Comment (dsifford) [inline middleware/package.json:18]: Removed this because we're not using it right now.

File: middleware/lib/middleware/auth.js (hunk 1 above)
Comment (dbartalino) [reply middleware/lib/middleware/auth.js:19]: You have the backslashes in each place you replaced the URL with the variable, so this should be fine.

**Review by dbartalino — APPROVED** [review]
Approved. Thanks for separating this out.

File: middleware/lib/middleware/auth.js (hunk 1 above)
Comment (dsifford) [reply middleware/lib/middleware/auth.js:19]: `issuer:` is missing a backslash where it had one before.

**Pull request merged**